src/Domain/Entity/RelatedLink.php

```diff
@@ -65,6 +66,45 @@ public function getType(): string
 
     public function isExternal(): bool
     {
+        if ($this->isExternal === null) {
+            $this->setUriMetadata();
+        }
+
+        // Assert to make PHPStan confident that setUriMetadata sets isExternal
+        // to a not-null value. Without this PHPStan raises a wanring
+        // See https://github.com/phpstan/phpstan/issues/856
+        assert($this->isExternal !== null);
```

Comment:
I really don't like having code in here just to please PHPStan. Can't you just 
uUse the magic file to make it shut up instead

Reply:
Why don't you set `$this->isExternal` in this method? And just get the other method to return it for you.

Reply:
@votemike `setUriMetadata()` sets both `$this->isExternal` and `$this->host`. Inlining the contents of `setUriMetadata()` into both `getHost()` and `isExternal()` works but the function is just big enough to make that duplication look weird. Having `setUriMetadata()` return an array of two values and doing `list($this->host, $this->isExternal) = $this->setUriMetadata();` in `getHost()` and `isExternal()` also does not work.

@BjornTwachtmann yep sure.

Reply:
I meant something like this:
```
public function isExternal(): bool
{
    if ($this->isExternal === null) {
        $hostLength = strlen($this->getHost());
        $this->isExternal = !(
            ($hostLength == 0) ||
            ($hostLength >= 9 && (strpos($this->host, 'bbc.co.uk', -9) !== false)) ||
            ($hostLength >= 7 && (strpos($this->host, 'bbc.com', -7) !== false))
        );
    }
    return $this->isExternal;
}

public function getHost(): string
{
    if ($this->host === null) {
        $matches = [];
        preg_match('@^(?:https?://)([^/?]+)@i', $this->uri, $matches);
        $this->host = $matches[1] ?? '';
    }
    return $this->host;
}
```

This means that you won't need PHPStan magic.
I also think it's clearer that the other way.